pkg/controller/cronjob/utils.go

```diff
@@ -187,10 +187,12 @@ func copyAnnotations(template *batchv1.JobTemplateSpec) labels.Set {
 // granularity of 1 minute for scheduling job.
 func getJobFromTemplate2(cj *batchv1.CronJob, scheduledTime time.Time) (*batchv1.Job, error) {
 	labels := copyLabels(&cj.Spec.JobTemplate)
-	annotations := copyAnnotations(&cj.Spec.JobTemplate)
 	// We want job names for a given nominal start time to have a deterministic name to avoid the same job being created twice
 	name := getJobName(cj, scheduledTime)
 
+	// Append job name to the job annotations.
+	annotations := appendJobCreationTimestamp(cj.CreationTimestamp.Time, copyAnnotations(&cj.Spec.JobTemplate))
```

Comment:
Do we want to use the scheduledTime as the String to the appended annotation?  This is also being set (appended) as the name from the getJobname which will ensure consistency.

Reply:
It might confuse others to use the scheduledTime instead, as we will append the expected/initial CreationTime.